### PR TITLE
Adds InputTableService support for blink tables

### LINKED ISSUE #4915: InputTableService support for blink tables
There isn't currently a way for clients to add data into a blink table.

It should be relatively easy to add support for this via the `io.deephaven.stream.TablePublisher` interface.

This feature _can_ be much more limited in scope than `MutableInputTable` suggests - clients only need the semantics offered via the InputTableService, which is simply:

```proto
service InputTableService {
    rpc AddTableToInputTable(AddTableRequest) returns (AddTableResponse) {}
   ...
}

message AddTableRequest {
    Ticket input_table = 1;
    Ticket table_to_add = 2;
}
```

### PR DESCRIPTION
**Pull request by devinrsmith**
This adds `TablePublisher#inputTable`, which adds a very simple `InputTableUpdater` implementation to the blink table.

Also, some further simplifications in the spirit of #4923: `InputTableUpdater#getDescription`, `InputTableUpdater#getTable`, and `InputTableUpdater#canEdit` were removed due to no usages.

Fixes #4915

### CODE REVIEW

File: engine/table/src/main/java/io/deephaven/stream/TablePublisher.java [hunk 1]
```diff
@@ -169,6 +169,27 @@ public boolean isAlive() {
...
 
+    /**
+     * The {@link Table#BLINK_TABLE_ATTRIBUTE blink table} with its {@link Table#getAttribute(String) attribute}
+     * {@value Table#INPUT_TABLE_ATTRIBUTE} set to an {@link io.deephaven.engine.util.input.InputTableUpdater}
+     * implementation based on {@code this}. This is primarily useful for existing code that already works with
+     * {@link io.deephaven.engine.util.input.InputTableUpdater} - new code should probably prefer to work directly with
+     * {@code this}.
+     *
+     * <p>
+     * May return {@code null} if invoked more than once and the initial caller does not enforce strong reachability of
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/stream/TablePublisher.java:180]: The `InputTableUpdater` you're providing doesn't match the usual semantics (blocking until published). You may want to document this.

File: qst/src/main/java/io/deephaven/qst/table/BlinkInputTable.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        return ImmutableBlinkInputTable.builder()
+                .schema(schema)
+                .build();
+    }
+
+    public abstract TableSchema schema();
+
+    @Default
+    UUID id() {
+        return UUID.randomUUID();
```
Comment (rcaudy) [inline qst/src/main/java/io/deephaven/qst/table/BlinkInputTable.java:30]: It's not obvious to me why this follows a different pattern than the other input tables.

Comment (devinrsmith) [reply qst/src/main/java/io/deephaven/qst/table/BlinkInputTable.java:30]: oversight on my part, will fix.

**Review by chipkent — APPROVED** [review]
Python looks fine.

**Review by rcaudy — COMMENTED** [review]
We want to update `InputTableUpdater.add` to explain that the semantics of "added" are implementation defined.

File: server/src/main/java/io/deephaven/server/table/inputtables/InputTableServiceGrpcImpl.java [hunk 3]
```diff
@@ -96,13 +97,18 @@ public void addTableToInputTable(
...
 
                         // actually add the tables contents
-                        try {
-                            inputTableUpdater.add(tableToAdd);
-                            GrpcUtil.safelyComplete(responseObserver, AddTableResponse.getDefaultInstance());
-                        } catch (IOException ioException) {
-                            throw Exceptions.statusRuntimeException(Code.DATA_LOSS,
-                                    "Error adding table to input table");
-                        }
+                        inputTableUpdater.addAsync(tableToAdd, new InputTableStatusListener() {
```
Comment (rcaudy) [inline server/src/main/java/io/deephaven/server/table/inputtables/InputTableServiceGrpcImpl.java:100]: Do we have to worry about the async calls throwing?

**Review by rcaudy — COMMENTED** [review]
.

File: server/src/main/java/io/deephaven/server/table/inputtables/InputTableServiceGrpcImpl.java (hunk 3 above)
Comment (devinrsmith) [reply server/src/main/java/io/deephaven/server/table/inputtables/InputTableServiceGrpcImpl.java:100]: No - the ExportBuilder has onError handler that plumbs any unexpected exceptions as errors to the response observer.

File: engine/table/src/main/java/io/deephaven/stream/TablePublisher.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 import io.deephaven.util.annotations.TestUseOnly;
 
 import javax.annotation.Nullable;
+import java.util.Collection;
```
Comment (rcaudy) [inline engine/table/src/main/java/io/deephaven/stream/TablePublisher.java:13]:
```suggestion
```

**Review by rcaudy — APPROVED** [review]

**Pull request merged**